noxfile.py

```diff
@@ -122,12 +127,13 @@ def tests(session, shard):
     default_test_dirs = [f"integrations/{shard}/"]
     test_dirs_dict = {
         "custom": [],
-        "trace": ["trace/"],
         "flow": ["flow/"],
         "trace_server": ["trace_server/"],
         "trace_server_bindings": ["trace_server_bindings"],
         "mistral": ["integrations/mistral/"],
         "scorers": ["scorers/"],
+        "trace": ["trace/"],
+        **{shard: ["trace/"] for shard in trace_shards},
```

Comment:
I know based on your metrics this isn't the case, but at first glance this looks like it would run the full trace suite for each shard. Could you add a comment clarifying what this is doing?